hometask-5

YuliyaBobrykgit opened the pull request:
filled script.js file

YuliyaBobrykgit committed:
hometask-5

filled script.js file

mdubik777 commented on script.js:
```diff
@@ -1,15 +1,70 @@
+function doForecast() {
+    if (currentForecastContainer.textContent.trim() === '') {
+        displayRandomForecast();
+        generateRandomProbability();
```
Юля,  хорошо,  но логически можно сделать лучше. У тебя 2 так называемые части: 1 - это генерация предсказания, 2я - это сохранение его в список. 
Вот и объедини displayRandomForecast();   и generateRandomProbability();  в одну, например 
displayForecast(); не нужно дробить и вызывать отдельно 2

YuliyaBobrykgit replied:
resolved

mdubik777 commented on script.js:
```diff
@@ -1,15 +1,70 @@
-/* При генерации нового предсказания старое предсказание должно добавляться в начало списка «Мои предсказания» — .forecasts  */
+        historyText.textContent = savedText;
+        historyProbability.textContent = savedProbability;
+        forecastItemTemplate.after(forecastHistoryContainer);
```
Это 2я часть  -  добавление в список. с 35й по 44ю  вынеси в отдельную функцию , например addForecastToList.  И тогда обработчик будет выглядеть как  

if (currentForecastContainer.textContent.trim() !== '') {
   addForecastToList();
}
displayForecast();

YuliyaBobrykgit replied:
resolved

YuliyaBobrykgit committed:
hometask-5-edited

merged generateRandomForecast(); displayRandomForecast(); generateRandomProbability() in one function displayForecast()

moved function saveForecastHistory() out of the final doForecast() function

mdubik777 commented on script.js:
```diff
@@ -1,15 +1,66 @@
+    } else {
+        saveForecastHistory();
+    }
+}
```
Тут не нужно else
if (currentForecastContainer.textContent.trim() !== '') {
addForecastToList();  - это выполнится при условии,  что уже есть сгенеренное предсказание
}
displayForecast();   - это выполнится всегда

YuliyaBobrykgit replied:
resolved

mdubik777 commented on script.js:
```diff
@@ -1,15 +1,66 @@
+    historyText.textContent = savedText;
+    historyProbability.textContent = savedProbability;
+    forecastItemTemplate.after(forecastHistoryContainer);
+    displayForecast();
```
Функция saveForecastHistory выполняем сохранение,  не нужно в ней вызывать другое действие.

YuliyaBobrykgit replied:
resolved

YuliyaBobrykgit committed:
hometask-5-edited(2)

removed odd function call inside saveForecastHistory();

edited displayForecast().
